packages/ui-test-locator/src/utils/locator.ts

```diff
@@ -45,26 +52,32 @@ export function locator(
   }
   queryAll.displayName = containerSelector
 
-  const query = (...args: any[]) => {
-    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-    // @ts-ignore
+  const query = (...args: QueryArguments) => {
+    // @ts-expect-error ...args should be refactored for better typing
     return firstOrNull(queryAll(...args))
   }
 
   const findAll = (...args: QueryArguments) => {
     const { element, selector, options } = parseQueryArguments(...args)
+    // this could be typed better. findAllByQuery should be generic, accepting
+    // customMethods as a generic param.
     return findAllByQuery(queryAll, element, selector, {
       ...options,
       customMethods: {
         ...customMethods,
         ...options.customMethods
       }
-    })
+      // Adding here the generic T argument makes TS able to autocomplete custom
+      // methods (if they are a compile time constant)
+      // Typing this properly would require to make QueryArguments generic,
+      // which would introduce a tons of complexity, so a refactor of this
+      // library would be preferred.
+    }) as Promise<QueriesHelpersEventsType[] & ObjWithCutFirstArg<T>>
```

Comment:
This makes it possible to use autocomplete with custom methods in locators.

Reply:
To be honest I'm not completely sure how this type works, as I see the Promise will contain an array and an object as well?
How that would look like in a real world example?

Reply:
hmm this was a bug, I fixed it